### ISSUE TITLE
drain option...

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#37]
Some times, when queues build up, you want to make a change to how the queues are defined.
To start fresh, one needs to create new queues with new bindings, but just doing a restart
will abandon the old queue. so would like to have an option to invoke the tools with the 'drain' argument. This will remove the queue bindings, so that the queue stops filling up, and can be drained.... so that no notifications are missed.

sr_subscribe --drain config start

wait until the queue is empty.

then stop it.
then change the config, and start without --drain, and will work normally.
.
.
.
with save/restore implemented. one could stop a consumer, note the queue (old_queue)
start a using a new queue (and new bindings.) Then start a shovel with -save -queue old_queue.
to obtain the messages from the old queue. 
sr_shovel -restore_queue new_queue 
will transfer the missing notifications to the new queue.
.
.
stopping the old queue, really means removing the binding on the old queue so that it stops getting messages.

 ... was: https://sourceforge.net/p/metpx/feature-requests/44/

I think most analysts would do this today using the rabbitmq management GUI which has substantially improved over the years.

**Comment by michelgrenier** [comment]
isn't this deprecated by retry ?

**Comment by petersilva** [comment]
I don´t think it is obsoleted by retry function... if someone want to change the expiry on a queue... or durable setting... how do they do it without data loss... one method would be to create a second queue with the same bindings, and once it is started, remove the bindings from the old queue, consume everything in the old queue, then keep receiving stuff from the new queue ... I dunno... 

I agree that this isn´t super important...
